Kudu.Core/Environment.cs

```diff
@@ -95,7 +96,17 @@ public class Environment : IEnvironment
             _siteExtensionSettingsPath = Path.Combine(SiteRootPath, Constants.SiteExtensionsCachePath);
             _diagnosticsPath = Path.Combine(SiteRootPath, Constants.DiagnosticsPath);
             _locksPath = Path.Combine(SiteRootPath, Constants.LocksPath);
-            _sshKeyPath = Path.Combine(rootPath, Constants.SSHKeyPath);
+
+            string siteName = System.Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
+            if (OSDetecter.IsOnWindows() || rootPath.TrimEnd(new char[] { '/', '\\' }).EndsWith(siteName, StringComparison.OrdinalIgnoreCase))
+            {
+                _sshKeyPath = Path.Combine(rootPath, Constants.SSHKeyPath);
+            }
+            else
+            {
+                // in linux, rootPath is "/home", while .ssh folder need to under "/home/{user}"
+                _sshKeyPath = Path.Combine(rootPath, System.Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME"), Constants.SSHKeyPath);
```

Comment:
WEBSITE_SITE_NAME is the same for all slots.   Do use `APP_POOL_ID = ~1suwatchslot01__fb0e` (and trim ~1 prefix) instead.